src/Compilers/VisualBasic/Portable/Binding/Binder.vb

```diff
@@ -452,6 +453,60 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Return type.GetUseSiteErrorInfo()
         End Function
 
+        Private Function GetInternalXmlHelperType(syntax As VisualBasicSyntaxNode, diagBag As DiagnosticBag) As NamedTypeSymbol
+            Debug.Assert(Me.ContainingType IsNot Nothing)
+            Dim containingAssembly = Me.ContainingType.ContainingAssembly
+
+            Dim typeSymbol = GetInternalXmlHelperType(containingAssembly)
+
+            Dim useSiteError = GetUseSiteErrorForWellKnownType(typeSymbol)
+            If useSiteError IsNot Nothing Then
+                ReportDiagnostic(diagBag, syntax, useSiteError)
+            End If
+
+            Return typeSymbol
+        End Function
+
+        Private Shared Function GetInternalXmlHelperType(containingAssembly As AssemblySymbol) As NamedTypeSymbol
+            Const metadataName = "My.InternalXmlHelper"
+
+            Dim typeSymbol = containingAssembly.GetTypeByMetadataName(
+                metadataName,
+                includeReferences:=True,
+                isWellKnownType:=True)  ' It's not, but we want the extra validation.
+
+            If typeSymbol Is Nothing Then
+                Dim emittedName = MetadataTypeName.FromFullName(metadataName)
+                typeSymbol = New MissingMetadataTypeSymbol.TopLevel(containingAssembly.Modules(0), emittedName)
+            End If
+
+            Return typeSymbol
+        End Function
+
+        Private Function GetInternalXmlHelperValueExtensionProperty() As PropertySymbol
+            Dim propertySymbol As PropertySymbol = Nothing
+            For Each candidate As Symbol In GetInternalXmlHelperType(Me.ContainingType.ContainingAssembly).GetMembers("Value")
+                If Not candidate.IsShared OrElse candidate.Kind <> SymbolKind.Property Then
+                    Continue For
+                End If
+
+                Dim candidateProperty = DirectCast(candidate, PropertySymbol)
+                If candidateProperty.Type.SpecialType <> SpecialType.System_String OrElse candidateProperty.ParameterCount <> 1 Then
+                    Continue For
+                End If
+
+                Dim parameterType = candidateProperty.Parameters(0).Type
+                If parameterType.OriginalDefinition.SpecialType <> SpecialType.System_Collections_Generic_IEnumerable_T OrElse
+                        DirectCast(parameterType, NamedTypeSymbol).TypeArgumentsNoUseSiteDiagnostics.Single() <> Me.Compilation.GetWellKnownType(WellKnownType.System_Xml_Linq_XElement) Then
+                    Continue For
+                End If
+
+                Return candidateProperty
```

Comment:
Do we care about ambiguity an case? Could it happen in metadata in presence of custom modifiers for example?